samples/emoji-search/android-composeui/src/main/kotlin/com/example/redwood/emojisearch/android/composeui/EmojiSearchActivity.kt

```diff
@@ -143,6 +148,27 @@ class EmojiSearchActivity : ComponentActivity() {
       manifestVerifier = ManifestVerifier.Companion.NO_SIGNATURE_CHECKS,
       embeddedFileSystem = applicationContext.assets.asFileSystem(),
       embeddedDir = "/".toPath(),
+      leakDetector = LeakDetector.timeBased(
+        timeSource = TimeSource.Monotonic,
+        leakThreshold = 10.seconds,
+        listener = object : LeakListener() {
+          override fun onReferenceLeaked(
```

Comment:
I’d like to somehow wire things together so that this is just another event to listen to in `EventListener`

Reply:
I originally had this design, but it meant that the ownership of creating the `LeakDetector` and running its coroutine are deep inside the code.

I think maybe the listener of whether something leaks should be supplied when you watch something. This allows the decision of whether the leak detector is real or not to be very far away from the "watch this and notify me" call.

Reply:
Once we get guest-side leaks, they'll come through the `EventListener`.